pkg/scheduler/framework/v1alpha1/types.go

```diff
@@ -455,24 +455,13 @@ func calculateResource(pod *v1.Pod) (res Resource, non0CPU int64, non0Mem int64)
 	resPtr := &res
 	for _, c := range pod.Spec.Containers {
 		resPtr.Add(c.Resources.Requests)
-
-		non0CPUReq, non0MemReq := schedutil.GetNonzeroRequests(&c.Resources.Requests)
-		non0CPU += non0CPUReq
-		non0Mem += non0MemReq
-		// No non-zero resources for GPUs or opaque resources.
 	}
```

Comment:
It seems we didn't take `initContainer` into consideration. Can you correct this?

BTW: there may still be some duplication code. We can refactor them later. In this PR, let's focus on fixing the bug.

Reply:
The duplication is in `fit.go`. I'm also ok with doing that in follow up PR so that we can cherrypick this in older versions more easily.